site-src/geps/gep-1867.md

```diff
@@ -0,0 +1,128 @@
+These higher level abstractions are handled by layered APIs (whether this is a CRD, an admission webhook, CI/CD tooling, etc).
+This allows users the flexibility to easily configure things per-pod basis.
+If the infrastructure admin wants to impose defaults or requirements on this flexibility, they are able to do so (in fact, `LimitRanger` provides a built in mechanism to do so).
+
+### Dynamic Changes
+
+Currently, `GatewayClass` is intended as a *template*.
+Changes to it are not expected to change deployed `Gateway`s.
+
+This makes usage problematic in a declarative way.
+For example, if I wanted to represent a `version` field and change that to trigger an upgrade, I would need to create an entirely new `Gateway`.
+
+## API
+
+In order to address the concerns above, I propose a standard `infrastructure` API is added to `Gateway` and `GatewayClass`.
+Note the important part of this is the `Gateway` change; the `GatewayClass` aspect is mostly for consistency.
+
+Note that many of the fields under `GatewayInfrastructure` refer to WIP GEPs.
+These fields are not tied to this GEP, and will only be included if the respective GEPs are approved.
+However, they are shown below to give a sense of the use cases the field can support.
+
+```go
+type GatewaySpec struct {
+  // Infrastructure defines infrastructure level attributes about this Gateway instance.
+  Infrastructure GatewayInfrastructure `json:"infrastructure"`
+  // ...
+}
+
+
+type GatewayClassSpec struct {
+  // Infrastructure defines infrastructure level attributes for all Gateways in this class.
+  // A Gateway may provide configuration for the same values; as all fields in GatewayInfrastructure are implementation specific,
+  // the merging logic between these is as well. However, the GatewayClass is generally expected to be providing defaults
+  // rather than overrides
+  Infrastructure GatewayInfrastructure `json:"infrastructure"`
+  // ...
+}
+
+type GatewayInfrastructure struct {
+  // AttachTo marks this Gateway as a child to the referenced parent Gateway.
+  // See GEP-1713 for details, which this depends on.
+  AttachTo LocalObjectReference
+
+  // Scope defines the "scope" of the Gateway's addresses.
+  // Valid options are ClusterLocal or Public
+  // See GEP-1651 for details, which this depends on.
+  // NOTE: that GEP is likely to be implemented in a different way, at which point this field can be removed from this struct.
+  Scope AddressScope
+
+  // Version defines the version of the Gateway infrastructure to use.
+  // This is an opaque string that is implementation specific.
+  // Some possible valid examples include "canary", "v1.2.3", "docker.io/example:latest".
+  Version string
+
+  // Size defines the size of the Gateway infrastructure.
+  // The meaning of this is implementation specific.
+  // GEP-1762 will recommend this to mean "Pod Replicas" for in-cluster deployments.
+  Size int
```

Comment:
I'm concerned whether an `int` can properly define the size of the Gateway infra. Size could define the type and amount of compute resources to use for the Gateway. IMO, `Replicas` is a well know definition throughout the industry and is a commonly requested config knob.

Reply:
Just a note - the specific fields of struct here are explicitly marked as not in scope for this GEP and are delegated to other GEPs. So your comment is valid but I don't think it should impact this PR at all. These are merely examples of the types of things that could be in the field.

That being said I will incorporate this into GEP-1762